### ISSUE TITLE
Revoke of a dertificate

### ISSUE THREAD

**Issue by tal-hason** [djkormo/adcs-issuer#93]
Hi, I am currently implementing the adcs plugin with cert-manager, and I can't see any reference to revoking certificates when deleting the certificate item from the cluster.

**Comment by djkormo** [comment]
What is your scenario ? You delete certificate object from kubernetes and certificate in ADCS is still active ?

**Comment by tal-hason** [comment]
Yes, after we removed the certificate item, we checked in the adcs and it was still present...

**Comment by djkormo** [comment]
The current version does not support revoking certificates.
1. Certificate object on kubernetes is controller by cert-manager itself. Adcs issuer is only using certificaterequest on its loop and creates adcsrequest as dedicated resource (one to one)  to store information about certificate id from adcs service. 
2. We can try to make implementation when certificaterequest and its child adcsrequest are deleted to prepare call to rekove certificate from adcs service.   I can prepare dedicated finalizer on adcsrequest objects to do it.  It could be an option controlled via the dedicated configmap. 
3.  I have to find you to revoke certificate via adcs api.

**Comment by tal-hason** [comment]
Thanks, on the update, that option will be a great feature.

I will try to look over the API of the adcs to see if can assist what needs to be done to revoke a certificate.

**Comment by djkormo** [comment]
https://learn.microsoft.com/en-us/answers/questions/1664586/can-i-revoke-an-ssl-certificate-using-api-calls-to

**Comment by tal-hason** [comment]
So, from the link you shared it looks like it's possible. If the current implementation is using the COM interface.

then you can use:

ICertAdmin::RevokeCertificate

COM interface to automate certificate

https://learn.microsoft.com/en-us/windows/win32/api/certadm/nf-certadm-icertadmin-revokecertificate

**Comment by tal-hason** [comment]
I don't remember if there is any location that holds the serial number of the created certificate, but it can be stored as annotation or in a field in the status.

**Comment by djkormo** [comment]
Here you have this field
https://github.com/djkormo/adcs-issuer/blob/master/api/v1/adcsrequest_types.go#L55

**Comment by djkormo** [comment]
To check if it is possible from the implementation
https://stackoverflow.com/questions/37781676/how-to-use-com-component-object-model-in-golang
